### DISCUSSION TITLE
有无 导出excel 的 实践?  支持列分组

### DISCUSSION THREAD

**Discussion by vaynevayne** [ant-design/ant-design#42248] [Q&A] [answered]
掘金找了好几个demo, 导出列分组时 就会有数据缺失 
探索不动了 有没有哪位老哥愿意分享一下

**Comment by yoyo837** [comment]
antd@3 + exceljs

<img width="1720" alt="image" src="https://github.com/ant-design/ant-design/assets/6134547/ddf756c8-34b5-40ce-b7cf-dc244deb77be">

导出:
[收入汇总_营销中心收入汇总_现金流水_20230510140706.xlsx](https://github.com/ant-design/ant-design/files/11438607/_._._20230510140706.xlsx)

**Reply by yoyo837** [reply]
导出内容完全复用antd table 的行列合并功能, 支持 column render 和 redux (render可能有redux读取操作)

**Reply by vaynevayne** [reply]
还支持render, 看来挺健壮的,  能分享源码吗?

**Reply by yoyo837** [reply]
可以打包部分内容给你看看, 仅供参考

**Reply by yoyo837** [reply, marked as answer]
发邮箱

**Reply by vaynevayne** [reply]
1848212420@qq.com

**Comment by vaynevayne** [comment]
<img width="725" alt="image" src="https://github.com/ant-design/ant-design/assets/124666577/87718aa5-3de3-47b5-ac9f-96681b4b69e5">

有点瑕疵, 在spw里编辑会出问题,我修复试试

**Reply by yoyo837** [reply]
列拉宽一点试试, 我这没问题

**Reply by yoyo837** [reply]
列宽等于 antd table column width

**Reply by vaynevayne** [reply]
这样啊 可以

**Reply by yoyo837** [reply]
反正就是和antd table本身的功能耦合, 这样, 开发人员在修改table列配置的时候, 导出的文件是一起变动的. 我们这边还做了个性化, 支持用户隐藏显示不关心的数据列, 拖拽换位置本身也要支持的太复杂就没启用, 导出的文件也会随个性化变动.

**Reply by vaynevayne** [reply]
隐藏显示列 你是columns受控还是内部状态?
列拖拽 你试试 https://github.com/raisezhang/react-drag-listview 
我打算后期支持 列拖出表格来隐藏列,
花样太多, 想封装一个hook来实现列显示隐藏

**Reply by yoyo837** [reply]
我这个稳定了, 不打算动了.

**Reply by yoyo837** [reply]
> 隐藏显示列 你是columns受控还是内部状态? 列拖拽 你试试 https://github.com/raisezhang/react-drag-listview 我打算后期支持 列拖出表格来隐藏列, 花样太多, 想封装一个hook来实现列显示隐藏

内部状态

**Comment by yoyo837** [comment]
<img width="1718" alt="image" src="https://github.com/ant-design/ant-design/assets/6134547/a536fb4e-ba06-442a-a38f-4bce5c3fc75c">

这个行列合并都有的
[收入汇总_综合收入汇总_消费收入_20230510143252.xlsx](https://github.com/ant-design/ant-design/files/11438772/_._._20230510143252.xlsx)

**Comment by vaynevayne** [comment]
https://github.com/exceljs/exceljs/pull/1889#issuecomment-1541671550

exceljs 要官方支持了

**Reply by yoyo837** [reply]
Marked
